File: jzoffer1-20/20.cpp
class Solution {
public:
    void push(int value) {
        stack1.push(value);
        if(stack2.empty())
        {
            stack2.push(value);
        }
        if(stack2.top()>value){
            stack2.push(value);
        }
    }
    void pop() {
        if(stack1.top()==stack2.top())
        {
            stack2.pop();
        }
        stack1.pop();
        
    }
    int top() {
        return stack1.top();
    }
    int min() {
        return stack2.top();
    }
private:
    stack<int> stack1;
    stack<int> stack2;
};

File: jzoffer1-20/20.js
var stack1=[];
function push(node)
{
    stack1.push(node);
}
function pop()
{
    stack1.pop();
}
function top()
{
    return stack1[stack1.length-1];
}
function min()
{
    return Math.min(...stack1);
    //return Math.min.apply(this.stack1);
}